collection.go

```diff
@@ -578,17 +579,55 @@ func (cl *collectionLoader) populateMaps() error {
 
 // resolveKconfig resolves all variables declared in .kconfig and populates
 // m.Contents. Does nothing if the given m.Contents is non-empty.
-func resolveKconfig(m *MapSpec) error {
+func resolveKconfig(m *MapSpec, additionalKconfig string) error {
 	ds, ok := m.Value.(*btf.Datasec)
 	if !ok {
 		return errors.New("map value is not a Datasec")
 	}
 
+	kernelRelease, err := internal.KernelRelease()
+	if err != nil {
+		return fmt.Errorf("cannot get kernel release: %w", err)
+	}
+
+	path := fmt.Sprintf("/boot/config-%s", kernelRelease)
+
+	config, err := kconfig.ParseKconfig(path, false)
+	if err != nil {
+		config, err = kconfig.ParseKconfig("/proc/config.gz", true)
+		if err != nil {
+			// If there is no file providing kconfig, we just default to empty one.
+			config = make(map[string]string)
+		}
+	}
+
+	if len(additionalKconfig) > 0 {
+		err := kconfig.PatchKconfig(config, additionalKconfig)
+		if err != nil {
+			return fmt.Errorf("cannot add kconfig patch: %w", err)
+		}
+	}
+
 	data := make([]byte, ds.Size)
 	for _, vsi := range ds.Vars {
 		v := vsi.Type.(*btf.Var)
+		n := v.TypeName()
+
+		if strings.HasPrefix(n, "CONFIG_") {
```

Comment:
Why do you allow kconfig to override "specials" like `LINUX_VERSION_CODE`? I would expect it the other way around.